auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, modules, _
+
+
+class AuditlogRule(models.Model):
+    _inherit = 'auditlog.rule'
+
+    field_ids = fields.Many2many(
+            'ir.model.fields',
+            required=True,)
+    group_ids = fields.Many2many(
+        'res.groups', 
+        string='Groups',
+        help="""Groups that will be allowed to see the logged fields, if left empty 
+                all groups will be allowed (global rule creation)"""
+    )
+
+    @api.multi
+    def write(self, vals):
+        res = super(AuditlogRule, self).write(vals)
+        for this in self:
+            if any([x in vals for x in ('group_ids', 'field_ids')]):
+                this.generate_rules()
+        return res
+
+    def generate_rules(self):
+        old_rule = self.env['ir.rule'].search([
+            ('auditlog_id', '=', self.id)])
+        # get all the ones generated by this module
+        # and wipe them out if not in the created group.
+        domain_force = "[ " + " ('log_id.model_id' , '=', %s)," % (
+                self.model_id.id) + ' '.join(
+                ["'|'," for field in self.field_ids[:-1]]) + ",".join(
+                ["('field_id', '=', %s)" % (
+                    field.id) for field in self.field_ids]) + "]"
+        values = ({             
+            'name': 'auditlog_extended_%s' % self.id,
+            'model_id': self.env.ref(
+                'auditlog.model_auditlog_log_line').id,
+            'groups': [(6, 0, self.group_ids.ids)],
+            'perm_read': True,
+            'domain_force': domain_force,
+            'auditlog_id': self.id,
+        })
```

Comment:
Refactor this part away into a `_prepare_rule_values` function, which can be overridden by submodules if needed.